src/phone_number/landline.rs

```diff
@@ -1,17 +1,42 @@
 use lazy_static::lazy_static;
 use regex::Regex;
 
+use crate::province::{IranProvince, PROVINCES};
+
 lazy_static! {
     static ref LANDLINE_NUMBER_REGEX: Regex =
         Regex::new(r"(\+98|0|98|0098)?([1-9]{2})+(\d{8})$").unwrap();
 }
-
-pub fn is_valid_landline_number<T: AsRef<str>>(number: T) -> bool {
-    LANDLINE_NUMBER_REGEX.is_match(number.as_ref())
+/// trait helper for validate landnumber and more ...
+pub trait LandlineNumber: AsRef<str> {
+    /// check landline number is valid number
+    fn is_valid_landline_number(&self) -> bool {
+        LANDLINE_NUMBER_REGEX.is_match(self.as_ref())
+    }
+    /// get three-digit prefix landline number
+    fn get_prefix_landline_number(&self) -> crate::Result<String> {
+        LANDLINE_NUMBER_REGEX
+            .captures(&self.as_ref())
+            .map(|c| format!("0{}", &c[2]))
+            .ok_or_else(|| "invalid landline number".into())
+    }
+    fn get_province_from_number(&self) -> crate::Result<Option<IranProvince>> {
+        self.get_prefix_landline_number()
+            .and_then(|p| {
```

Comment:
There is a clippy message about this, run `make check`:
ref: https://rust-lang.github.io/rust-clippy/master/index.html#bind_instead_of_map